test/helpers

```diff
@@ -23,3 +23,55 @@ test::helpers::get_php_version() {
     HOME="${old_home}"
     export HOME
 }
+
+test::helpers::list_php_modules() {
+    local old_home
+
+    old_home="${HOME}"
+
+    HOME="/app"
+    export HOME
+
+    pushd "${HOME}" > /dev/null
+
+    cp --archive --recursive "${BUILD_DIR}"/* "${HOME}/"
```

Comment:
I agree... but couldn't find something "ideal".

The pattern is always:
1. Copy the result of `compile` in `/app`, so we know we work with the result of `compile`. Adjust the environment.
2. Call something and capture the output (the _something_ we want to test).
3. Clean so the next tests are done in iso-conditions.

I'd like to make sure we can't call 1. without calling 3. atfer.

I'll see what I can do and come up with a suggestion :)

Reply:
This lead me to a rather big change in the code organization... and, consequently, a pretty big review for you. Sorry :yum:
 
More seriously, I hesitated a lot between two approaches:

1. The first approach is closer to the unit testing philosophy: every little thing we test should be in its own test function. This means we have to call 'compile' quite a lot of times (for almost no benefit IMHO).
2. The second one is more economic: we reproduce the deployment step by step and test everything in between (see `test::helpers::test_deploy`).

I finally opted for the 2nd approach, mainly because it doesn't require to re-compile everything again and again.
**But maybe the 1st would have been better?**